### PR TITLE
PROD-890 #1822 - Fix: The profile completion widget is not updating after profile type was changed

### LINKED ISSUE #1822: The profile completion widget is not updating after profile type was changed
**To Reproduce**
1. Create at least two profile types.
2. Then, create some profile fields for each profile type.
3. Use the Profile Completion widget on one of your pages.
4. Now, assign one of your profile types to one of your users.
5. Login using that user and try to complete all the requirements in the Profile Completion widget.
6. After that, change the profile type of that user.
7. Notice that the Profile Completion widget remains completed even there are profile fields that need to be completed.

**Expected Behavior**
The status of the Profile Completion widget should also be updated once the profile type has been changed. This issue affects those sites with a Membership system, wherein a user upgraded to a certain profile but their profile completion widget remains outdated.

**Screencast**
https://cutt.ly/Ship1vP

**Support ticket links**
https://secure.helpscout.net/conversation/1344210577/109839

**Jira issue** : [PROD-890]

[PROD-890]: https://buddyboss.atlassian.net/browse/PROD-890?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by milanhirapra**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1822 .

### How to test the changes in this Pull Request:

1. Create at least two profile types.
2. Then, create some profile fields for each profile type.
3. Use the Profile Completion widget on one of your pages.
4. Now, assign one of your profile types to one of your users.
5. Login using that user and try to complete all the requirements in the Profile Completion widget.
6. After that, change the profile type of that user.
7. Profile Completion widget updated.

### Proof Screenshots or Video
https://vimeo.com/588860417/6cf0eb518e

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-core/bp-core-functions.php [hunk 1]
```diff
@@ -5238,7 +5238,11 @@ function bp_core_profile_completion_steps_options() {
  *
  * @since BuddyBoss 1.4.9
  */
-function bp_core_xprofile_update_profile_completion_user_progress() {
+function bp_core_xprofile_update_profile_completion_user_progress( $user_id = '', $posted_field_ids = array(), $errors = array(), $old_values = array(), $new_values = array() ) {
+
+	if( empty( $user_id ) ){
+		$user_id 		= get_current_user_id();
```
Comment (dipskakadiya) [inline src/bp-core/bp-core-functions.php:5244]: @milanhirapra Check line no 5247 [ we already assigning current user-id to $user_id  ], We have duplicate code. the solution is not correct.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-core/bp-core-functions.php (hunk 1 above)
Comment (milanhirapra) [reply src/bp-core/bp-core-functions.php:5244]: @dipskakadiya I forgot to remove this line but, I added the condition if the user id is null then assign the current user id.

**Comment by milanhirapra** [comment]
@dipskakadiya I created another pull request as per your review.

**Review by dipskakadiya — APPROVED** [review]

File: src/bp-core/bp-core-functions.php [hunk 2]
```diff
@@ -5249,9 +5249,12 @@ function bp_core_profile_completion_steps_options() {
...
+	}
```
Comment (KartikSuthar) [inline src/bp-core/bp-core-functions.php:5256]: @milanhirapra please follow the phpcs here.

File: src/bp-xprofile/bp-xprofile-filters.php [hunk 3]
```diff
@@ -105,7 +105,7 @@
 add_action( 'xprofile_cover_image_uploaded', 'bp_core_xprofile_update_profile_completion_user_progress' ); // When cover photo uploaded from profile in Frontend.
 add_action( 'bp_core_delete_existing_avatar', 'bp_core_xprofile_update_profile_completion_user_progress' ); // When profile photo deleted from profile in Frontend.
 add_action( 'xprofile_cover_image_deleted', 'bp_core_xprofile_update_profile_completion_user_progress' ); // When cover photo deleted from profile in Frontend.
-add_action( 'xprofile_updated_profile', 'bp_core_xprofile_update_profile_completion_user_progress' ); // On Profile updated from frontend.
+add_action( 'xprofile_updated_profile', 'bp_core_xprofile_update_profile_completion_user_progress', 1, 5 ); // On Profile updated from frontend.
```
Comment (KartikSuthar) [inline src/bp-xprofile/bp-xprofile-filters.php:108]:
```suggestion
add_action( 'xprofile_updated_profile', 'bp_core_xprofile_update_profile_completion_user_progress', 20, 5 ); // On Profile updated from frontend.
```

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

**Review by chetansatasiya — COMMENTED** [review]
LGTM 🚀

**Review by chetansatasiya — APPROVED** [review]

**Pull request merged**